chain/mempool.go

```diff
@@ -13,9 +13,10 @@ import (
 )
 
 const (
-	// txNotFoundErr is an error returned from bitcoind's
+	// errTxNotFoundCode is the error code returned from bitcoind's
 	// `getrawtransaction` RPC when the requested txid cannot be found.
-	txNotFoundErr = "-5: No such mempool or blockchain transaction"
+	// https://github.com/bitcoin/bitcoin/blob/fa05a726c225dc65dee79367bb67f099ae4f99e6/src/rpc/rawtransaction.cpp#L366
+	errTxNotFoundCode = "-5"
```

Comment:
I'm not sure if it's safe to only look at the error code.
For example [here](https://github.com/bitcoin/bitcoin/blob/fa05a726c225dc65dee79367bb67f099ae4f99e6/src/rpc/rawtransaction.cpp#L356) we also get `-5` but with a completely different cause.
Sure, the result for us is just whether we log it as error or debug entry now. But if we change this should we make it a bit more robust? For example the `-5: no such mempool` substring (combined with a case insensitive comparison) would result in the originally intended behavior (I think).

Reply:
Good catch, I think your point is valid and I will change.